includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -1685,6 +1685,23 @@ public function capture_charge( $order ) {
 		$currency                 = WC_Payments_Utils::get_order_intent_currency( $order );
 
 		try {
+			$intent_id    = $order->get_transaction_id();
+			$intent       = $this->payments_api_client->get_intent( $intent_id );
+			$payment_type = $this->is_payment_recurring( $order->get_id() ) ? Payment_Type::RECURRING() : Payment_Type::SINGLE();
+
+			$metadata_from_intent = $intent->get_metadata(); // mobile app may have set metadata.
+			$metadata_from_order  = $this->get_metadata_from_order( $order, $payment_type );
+			$merged_metadata      = array_merge( $metadata_from_order, $metadata_from_intent ); // prioritize metadata from mobile app.
+
+			$this->payments_api_client->update_intention_unrestricted(
+				$intent_id,
+				null,
+				null,
+				null,
+				null,
+				$merged_metadata
+			);
+
 			$intent = $this->payments_api_client->capture_intention(
 				$order->get_transaction_id(),
```

Comment:
In the current implementation you have it in the `$intent_id`

Reply:
Done ✅ Thanks for spotting it.